pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -0,0 +1,720 @@
+		_, err := c.capiClient.ClusterV1alpha1().MachineSets(updatedMachineSet.Namespace).Update(updatedMachineSet)
+		if err != nil {
+			return fmt.Errorf("error updating machineset %s/%s: %v", machineSet.Namespace, machineSet.Name, err)
+		}
+	}
+	return nil
+}
+
+func buildMasterMachineSet(clusterDeployment *clustop.ClusterDeployment, clusterVersion *clustop.ClusterVersion) (*capi.MachineSet, error) {
+	machineSetConfig := masterMachineSetConfig(clusterDeployment)
+	if machineSetConfig == nil {
+		return nil, fmt.Errorf("cluster deployment %s/%s does not have a master machine set config", clusterDeployment.Namespace, clusterDeployment.Name)
+	}
+
+	machineSet := &capi.MachineSet{}
+	machineSet.Name = masterMachineSetName(clusterDeployment.Name)
+	machineSet.Namespace = clusterDeployment.Namespace
+	machineSet.Labels = clusterDeployment.Labels
+	if machineSet.Labels == nil {
+		machineSet.Labels = make(map[string]string)
+	}
+	machineSet.Labels[clusterDeploymentLabel] = clusterDeployment.Name
+	machineSet.OwnerReferences = []metav1.OwnerReference{*metav1.NewControllerRef(clusterDeployment, controllerKind)}
+	machineSetLabels := map[string]string{"machineset": machineSet.Name}
+	machineSet.Spec.Selector.MatchLabels = machineSetLabels
+	replicas := int32(machineSetConfig.Size)
+	machineSet.Spec.Replicas = &replicas
+	machineSet.Spec.Template.Labels = machineSetLabels
+	machineSet.Spec.Template.Spec.Labels = machineSetLabels
+
+	providerConfig, err := controller.MachineProviderConfigFromMachineSetConfig(machineSetConfig, &clusterDeployment.Spec, clusterVersion)
+	if err != nil {
+		return nil, err
+	}
+	machineSet.Spec.Template.Spec.ProviderConfig.Value = providerConfig
+	return machineSet, nil
+}
+
+func buildCluster(clusterDeployment *clustop.ClusterDeployment) (*capi.Cluster, error) {
+	cluster := &capi.Cluster{}
+	cluster.Name = clusterDeployment.Name
+	cluster.Labels = clusterDeployment.Labels
+	cluster.Namespace = clusterDeployment.Namespace
+	if cluster.Labels == nil {
+		cluster.Labels = make(map[string]string)
+	}
+	cluster.Labels[clusterDeploymentLabel] = clusterDeployment.Name
+	cluster.OwnerReferences = []metav1.OwnerReference{*metav1.NewControllerRef(clusterDeployment, controllerKind)}
+	providerConfig, err := controller.ClusterProviderConfigSpecFromClusterDeploymentSpec(&clusterDeployment.Spec)
+	if err != nil {
+		return nil, err
+	}
+	cluster.Spec.ProviderConfig.Value = providerConfig
+
+	// Set networking defaults
+	cluster.Spec.ClusterNetwork.ServiceDomain = defaultServiceDomain
+	cluster.Spec.ClusterNetwork.Pods.CIDRBlocks = []string{defaultPodCIDR}
+	cluster.Spec.ClusterNetwork.Services.CIDRBlocks = []string{defaultServiceCIDR}
+	return cluster, nil
+}
+
+// validateAWSRegion will ensure the cluster's version has an AMI defined for its region. If not a condition will
+// be added indicating the problem. Similarly if the problem clears up, the condition will be updated.
```

Comment:
Almost surely this is my bad but most of these comments state things that don't happen here, they happen up in the caller. Looks like the method is just validating and returning an error if there's a problem.